File: Problems/1448_Count_Good_Nodes_in_Binary_Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int goodNodes(TreeNode* root) {
        int cnt = 0;
        traverse(root, INT_MIN, cnt);
        return cnt;
    }
    void traverse(TreeNode* node, int seenBig, int &cnt) {
        if (!node) return;
        if (node -> val >= seenBig) {
            ++cnt;
        }
        seenBig = max(seenBig, node -> val);
        traverse(node -> left, seenBig, cnt);
        traverse(node -> right, seenBig, cnt);
    }
};

File: Problems/1448_Count_Good_Nodes_in_Binary_Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        cnt = 0
        
        def traverse(node, seenBig):
            if not node:
                return
            if node.val >= seenBig:
                nonlocal cnt
                cnt += 1
            seenBig = max(seenBig, node.val)
            traverse(node.left, seenBig)
            traverse(node.right, seenBig)
        
        traverse(root, -10**4-1)
        return cnt